### ISSUE TITLE
Body parser maps files to params, but some time in the future

### ISSUE THREAD

**Issue by rehia** [restify/node-restify#847]
I was trying to use Restify behavior to get an uploaded file in `request.params` using this:
```javascript
server.use(restify.bodyParser({
    mapParams: true,
    mapFiles: true,
    keepExtensions: true,
    uploadDir: '/tmp/'
}));
```
I have this endpoint configured:
```javascript
server.post('/file', function (request, response, next) {
    console.log(request.params);
    console.log(request.files);
    response.send('OK');
    return next();
});
```
When I upload a file, it appears that `request.files` contains the file, but `request.params` is **empty** !
After a debugging session, I found in `multipart_parser.js`, files a read and mapped to `request.params` asynchronously.
I finally tries to execute the statements of the request handler in a `setTimeout`.
And I finally got my file mapped in `request.params` !
```javascript
setTimeout(function () {
    console.log(request.params);
    console.log(request.files);
    response.send('OK');
    return next();
}, 100);
```

What can we do about this situation, so that when my handler has the control, files are set in `request.params` ?

Thanks

**Comment by yunong** [comment]
@rehia This is a bug in the multipart_parser.js as part of commit c0d1dd40. Definitely a bug here. I'll make sure this gets fixed as part of the upcoming release.

**Comment by rehia** [comment]
@yunong OK thanks. The comment of the commit is right, in my opinion.

**Comment by yunong** [comment]
@rehia Can you give me an example curl request that I can use to test a fix?

**Comment by rehia** [comment]
@yunong unfortunately, I don't have one. I was just trying on my computer. Nothing available.

**Comment by yunong** [comment]
This has been fixed as per `c604cd8`. If you pick up #master you'll get the fix right away.

**Comment by rehia** [comment]
@yunong OK thanks !
